src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala

```diff
@@ -324,6 +324,31 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def getBucketUsage(bucketName: String, maxResults: Option[Long]): Future[BigInt] = {
+    def sumItemSizes(items: java.util.List[StorageObject]) = {
+      (BigInt(0) /: items) { (sum, o) => sum + o.getSize }
```

Comment:
Thanks. In this case, I think I need to `foldLeft` or `foldRight`:

    items.foldLeft(BigInt(0)) { (sum, o) => sum + o.getSize }

The reason is that `items` is an array of some object type rather than an array of `BigInt`. Is that okay or should I first `map` it to an array of `BigInt` and then `fold` it?